minvws/nl-kat-coordination
# Fix empty consumes of boefjes will trigger tasks in scheduler

### Changes

When a boefje gets enabled the scheduler will check whether this boefje can run on ooi's that allow this boefje to run on. To do that the scheduler checks what types a boefje can consume. The scheduler will ask octopoes to return those ooi's. When the `consumes` field of a boefje is empty it will ask octopoes to return ooi's and it will do so when an empty list is provided.

This PR adds a check before the scheduler will ask octopoes to return ooi's, it will check if the `consumes` field is empty or not and will not request octopoes for ooi's when it is.

### Issue link

Closes [#3015](https://github.com/minvws/nl-kat-coordination/issues/3015)

### QA steps

From what I can see it is not possible to set the consumes of a boefje to an empty list from the webapp. In order to make sure that everything should be working as intended, is to enable a boefje and make sure that the types the ooi's consumes, tasks are being created for ooi's of that type.
